File: 2831-number-of-beautiful-pairs/number-of-beautiful-pairs.java
class Solution {
    public int gcd(int a,int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    public int countBeautifulPairs(int[] nums) {
        int c=0;
        for(int i=0;i<nums.length;i++){
            for(int j=i+1;j<nums.length;j++){
                if(gcd(Integer.valueOf(String.valueOf(nums[i]).charAt(0)+""),nums[j]%10)==1) c++;
            }
        }
        return c;
    }
}

File: 2831-number-of-beautiful-pairs/number-of-beautiful-pairs.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function gcd($a,$b){
        return $b==0?$a:$this->gcd($b,$a%$b);
    }
    function countBeautifulPairs($nums) {
        $c=0;
        for($i=0;$i<count($nums);$i++){
            $first=intval(strval($nums[$i])[0]);
            for($j=$i+1;$j<count($nums);$j++){
                if($this->gcd($first,$nums[$j]%10)==1) $c++;
            }
        }
        return $c;
    }
}